### PR TITLE
make use of virtual database threading instead of using explicit star…

### PR DESCRIPTION
**Pull request by ashley-taylor**
…t for all requests

### CODE REVIEW

File: graphql-database-manager-core/src/main/java/com/fleetpin/graphql/database/manager/Database.java [hunk 1]
```diff
@@ -311,31 +320,14 @@ private <T> CompletableFuture<List<T>> merge(Stream<CompletableFuture<T>> stream
...
-
-	void start() {
+	private void start() {
 		if (items.dispatchDepth() > 0 || queries.dispatchDepth() > 0 || queryHistories.dispatchDepth() > 0 || put.dispatchSize() > 0) {
 			CompletableFuture<?>[] all = new CompletableFuture[] { items.dispatch(), queries.dispatch(), queryHistories.dispatch(), put.dispatch() };
-			CompletableFuture.allOf(all).join();
+			try {
+				CompletableFuture.allOf(all).join();
+			} catch (Exception e) {
+				e.printStackTrace();
```
Comment (ashley-taylor) [inline graphql-database-manager-core/src/main/java/com/fleetpin/graphql/database/manager/Database.java:329]: forgot to delete

File: graphql-database-manager-core/src/main/java/com/fleetpin/graphql/database/manager/Database.java [hunk 2]
```diff
@@ -423,4 +415,36 @@ public Set<String> getLinkIds(Table entity, Class<? extends Table> type) {
 		}
 		return Collections.unmodifiableSet(links);
 	}
+
+	public void handleFuture(CompletableFuture<?> future) {
```
Comment (ashley-taylor) [inline graphql-database-manager-core/src/main/java/com/fleetpin/graphql/database/manager/Database.java:419]: make private

File: graphql-database-manager-core/src/main/java/com/fleetpin/graphql/database/manager/VirtualDatabase.java [hunk 3]
```diff
@@ -17,195 +17,137 @@
...
-		return handleFuture(future);
+		return database.takeHistoryBackup(organisationId).join();
 	}
 
 	public <T extends Table> T takeHistoryBackup(final T entity, final Class<? extends Table> type, final String targetId) {
-		var future = database.unlink(entity, type, targetId);
-		return handleFuture(future);
-	}
-
-	public <T> T handleFuture(CompletableFuture<T> future) {
```
Comment (Konyx) [inline graphql-database-manager-core/src/main/java/com/fleetpin/graphql/database/manager/VirtualDatabase.java:183]: Breaking change

File: graphql-database-manager-test/src/test/java/com/fleetpin/graphql/database/manager/test/DynamoDbIndexesTest.java [hunk 4]
```diff
@@ -74,15 +73,19 @@ void testGlobalInheritance(@DatabaseNames({ "prod", "stage" }) final Database db
 	@TestDatabase
 	void testSecondary(final Database db) throws InterruptedException, ExecutionException {
 		var list = db.querySecondary(SimpleTable.class, "garry").get();
+		System.out.println("123");
```
Comment (Konyx) [inline graphql-database-manager-test/src/test/java/com/fleetpin/graphql/database/manager/test/DynamoDbIndexesTest.java:76]: remove

File: graphql-database-manager-test/src/main/java/com/fleetpin/graphql/database/manager/test/DynamoDbInitializer.java [hunk 5]
```diff
@@ -209,33 +213,33 @@ static DynamoDbManager getDatabaseManager(
...
-	//                .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
-	//                .registerModule(new ParameterNamesModule())
-	//                .registerModule(new Jdk8Module())
-	//                .registerModule(new JavaTimeModule())
-	//                .disable(DeserializationFeature.READ_DATE_TIMESTAMPS_AS_NANOSECONDS)
-	//                .disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
-	//                .disable(SerializationFeature.WRITE_DATE_TIMESTAMPS_AS_NANOSECONDS)
-	//                .disable(SerializationFeature.WRITE_DURATIONS_AS_TIMESTAMPS)
-	//                .setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
+	// static Database getInMemoryDatabase(
```
Comment (Konyx) [inline graphql-database-manager-test/src/main/java/com/fleetpin/graphql/database/manager/test/DynamoDbInitializer.java:216]: Can we remove?

**Review by Konyx — CHANGES_REQUESTED** [review]
minor

**Review by Konyx — APPROVED** [review]

**Pull request merged**